packages/template-retail-react-app/app/components/store-locator-modal/index.jsx

```diff
@@ -5,89 +5,181 @@
+            )
+    }
+    const [userHasSetManualGeolocation, setUserHasSetManualGeolocation] = useState(
+        defaultUserHasSetManualGeolocation
+    )
 
-    const [searchStoresParams, setSearchStoresParams] = useState({
-        countryCode: DEFAULT_STORE_LOCATOR_COUNTRY_CODE,
-        postalCode: DEFAULT_STORE_LOCATOR_POSTAL_CODE
-    })
-
+    const [searchStoresParams, setSearchStoresParams] = useState(defaultSearchStoresParams)
     const form = useForm({
         mode: 'onChange',
         reValidateMode: 'onChange',
         defaultValues: {
-            countryCode: searchStoresParams.countryCode,
-            postalCode: searchStoresParams.postalCode
+            countryCode: userHasSetManualGeolocation ? searchStoresParams.countryCode : '',
+            postalCode: userHasSetManualGeolocation ? searchStoresParams.postalCode : ''
         }
     })
+
+    const searchStoresDataRef = useRef({
+        data: []
+    })
+
     var {data: searchStoresData, isLoading} = useSearchStores({
         parameters: {
-            countryCode: searchStoresParams.countryCode,
-            postalCode: searchStoresParams.postalCode,
+            countryCode: searchStoresParams.latitude ? undefined : searchStoresParams.countryCode,
+            postalCode: searchStoresParams.latitude ? undefined : searchStoresParams.postalCode,
+            latitude: searchStoresParams.countryCode ? undefined : searchStoresParams.latitude,
+            longitude: searchStoresParams.countryCode ? undefined : searchStoresParams.longitude,
             locale: intl.locale,
-            maxDistance: STORE_LOCATOR_DISTANCE
+            maxDistance: STORE_LOCATOR_DISTANCE,
+            limit: searchStoresParams.limit,
+            offset: 0
         }
     })
-    const storesInfo = isLoading ? undefined : searchStoresData?.data || []
+
+    if (isLoading === false) searchStoresDataRef.current = searchStoresData
+    const storesInfo = searchStoresDataRef.current.data ? searchStoresDataRef.current.data : []
+    const numStores = isLoading ? 0 : searchStoresData.total
 
     const submitForm = async (formData) => {
         const {postalCode, countryCode} = formData
         setSearchStoresParams({
             postalCode: postalCode,
-            countryCode: countryCode
+            countryCode: countryCode,
+            limit: 10
         })
+        setUserHasSetManualGeolocation(true)
     }
 
     const isDesktopView = useBreakpointValue({base: false, lg: true})
+    useEffect(() => {
+        if (!isServer)
+            window.localStorage.setItem(
+                'STORE_LOCATOR_SEARCH_STORES_PARAMS',
+                JSON.stringify(searchStoresParams)
+            )
+    }, [searchStoresParams])
+
+    useEffect(() => {
+        if (!isServer)
+            window.localStorage.setItem(
+                'STORE_LOCATOR_USER_HAS_SET_GEOLOCATION',
+                userHasSetManualGeolocation ? 1 : 0
+            )
+    }, [userHasSetManualGeolocation])
```

Comment:
useEffect does not run on the server, so `if (!isServer)` is unnecessary

Reply:
Thank you!